packages/manager/modules/sharepoint/src/sharepoint.service.js

```diff
@@ -623,32 +625,47 @@ export default class MicrosoftSharepointLicenseService {
     );
   }
 
-  getExchangeServices() {
-    return this.OvhApiEmailExchange.service()
-      .v7()
+  getAllExchangeServices() {
+    return this.OvhHttp
+      .get('/email/exchange', {
+        rootPath: 'apiv6',
+      })
+      .then((serviceIds) => this.$q.all(map(serviceIds, (serviceId) =>
+          this.getExchangeServices(serviceId)
+        )),
+      )
+      .then((services) => reduce(services, (flattened, other) => {
+          return flattened.concat(other);
+        }, []),
+      );
+  }
+
+  getExchangeServices(organizationName) {
+    return this.iceberg('/email/exchange/:organizationName/service')
```

Comment:
Nice job ! But what bothers me is that we've replaced one call with N+1 one, this might have quite some performance issue if there are quite a lot of services

Reply:
that's true

Reply:
I do agree. Since it has been validated as is it could be part of the next release. We will keep an eye on that to make sure that performance are not impacted.